omegaconf/_utils.py

```diff
@@ -635,32 +624,22 @@ def is_dict_annotation(type_: Any) -> bool:
     origin = getattr(type_, "__origin__", None)
     # type_dict is a bit hard to detect.
     # this support is tentative, if it eventually causes issues in other areas it may be dropped.
-    if sys.version_info < (3, 7, 0):  # pragma: no cover
-        typed_dict = hasattr(type_, "__base__") and type_.__base__ == Dict
-        return origin is Dict or type_ is Dict or typed_dict
-    else:  # pragma: no cover
-        typed_dict = hasattr(type_, "__base__") and type_.__base__ == dict
-        return origin is dict or typed_dict
```

Comment:
This gets into actual cleanup of 3.6 code.
Jasha, did we officially removed support for it yet?

Reply:
> This gets into actual cleanup of 3.6 code.

When you said "lgtm" I thought you were saying "yes" to my question "thoughts on enforcing Python >= 3.8?"

> @Jasha10 , did we officially removed support for it yet?

All official releases of OmegaConf / Hydra still support Python 3.6. I'm proposing to drop this support in the next release (because it will make our lives easier, and I doubt anyone still runing 3.6/3.7 would care about not being able to run the latest version).